src/assets/content.js

```diff
@@ -5,13 +5,56 @@
 // from the contents of the active page
 chrome.runtime.onMessage.addListener( function(request, sender, sendResponse) {
   if (request.message === 'send links') {
-    sendResponse(Array.from(document.links).map( function(val) {
-      return {
-        'href': val.href,
-        'host': val.host,
-        'tagName': val.tagName,
-        'hidden': val.hidden
-      };
-    }));
+    sendResponse(findLinks());
   }
 });
+
+function findLinks() {
```

Comment:
Can we create a separate js file for this? Does it needs to be in content.js?

Reply:
I think we can label multiple files as content script, so yeah it seems doable!

Reply:
Actually looks kind of tricky, I can't quite figure out how to reference another file, especially if it still needs to have access to the web page's DOM. It looks like this post had to do with a similar issue, but the workaround seems like more trouble than its worth. I wonder if this has changed at all since this was posted .

https://stackoverflow.com/questions/48104433/how-to-import-es6-modules-in-content-script-for-chrome-extension